### PR TITLE
Ci improvements

### PR DESCRIPTION
**Pull request by 0xtekgrinder**
Clean and improve the ci by removing useless node_modules, adding a more robust ci with slither and a coverage badge

### CODE REVIEW

File: .github/workflows/ci-deep.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  schedule:
+    - cron: "0 3 * * 0" # at 3:00am UTC every Sunday
+  workflow_dispatch:
+    inputs:
+      fuzzRuns:
+        default: "100000"
+        description: "Unit: number of fuzz runs."
+        required: false
+      invariantRuns:
+        default: "64"
```
Comment (GuillaumeNervoXS) [inline .github/workflows/ci-deep.yml:16]: why did you set to 64 and 2048? 
I think I would prefer more runs than depth, but if you have goo reasons let me know

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,104 +1,225 @@
...
-        uses: Rubilmax/foundry-gas-diff@v3.13.4
-        id: gas_diff
+      - name: Check storage layout
+        uses: Rubilmax/foundry-storage-check@v3.7
         with:
-          match: contracts/transmuter/DiamondProxy.sol
-          report: ethereum.gasreport.ansi
+          contract: contracts/transmuter/Layout.sol:Layout
+          rpcUrl: ${{ secrets.ETH_NODE_URI_MAINNET }}
+          address: "0x00253582b2a3FE112feEC532221d9708c64cEFAb"
```
Comment (GuillaumeNervoXS) [inline .github/workflows/ci.yml:224]: so you are using the on chain address, why would it fail the 1st time then?

File: .github/workflows/ci-deep.yml (hunk 1 above)
Comment (0xtekgrinder) [reply .github/workflows/ci-deep.yml:16]: It was a pseudo random value so let me know what values would you prefer

File: .github/workflows/ci.yml (hunk 2 above)
Comment (0xtekgrinder) [reply .github/workflows/ci.yml:224]: The address is an optional check as it will first compare it to the previous report

Comment (GuillaumeNervoXS) [reply .github/workflows/ci.yml:224]: Ok so it both check the previous and the on chain? Can you walk me through in vocal? Thanks

File: .github/workflows/ci-deep.yml (hunk 1 above)
Comment (GuillaumeNervoXS) [reply .github/workflows/ci-deep.yml:16]: Ok good then let's go with 300 runs and 50 depth

**Review by GuillaumeNervoXS — APPROVED** [review]

File: test/mock/MockLib.sol [hunk 3]
```diff
@@ -27,6 +29,10 @@ contract MockLib {
         return LibHelpers.piecewiseLinear(x, xArray, yArray);
     }
 
+    function implementationStorage() external pure returns (ImplementationStorage memory) {
+        return LibStorage.implementationStorage();
```
Comment (GuillaumeNervoXS) [inline test/mock/MockLib.sol:33]: Can we test what we get from that?

**Review by GuillaumeNervoXS — APPROVED** [review]

**Pull request merged**